backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/predicate/impl/ElasticsearchSearchPredicate.java

```diff
@@ -24,8 +25,14 @@ public interface ElasticsearchSearchPredicate extends SearchPredicate {
 
 	void checkNestableWithin(String expectedParentNestedPath);
 
+	default ElasticsearchSearchPredicate checkAcceptableAsBoolPredicateClause(String clauseType) {
+		return this;
+	}
+
 	JsonObject toJsonQuery(PredicateRequestContext context);
 
+	JsonElement toJsonKnn(PredicateRequestContext rootPredicateContext);
```

Comment:
Ah, so it wasn't possible to just add an `addKnn` method to `PredicateRequestContext` as we discussed?

Reply:
I wasn't sure how it all will play out so I went with the most straightforward thing. I've moved things around and this method is gone now.